#  Изучение развития игровой индустрии с 2000 по 2013 год

- Автор: Жигарева Виктория
- Дата: 28.01.2026

### Цели и задачи проекта

Цель проекта: изучить развитие игровой индустрии с 2000 по 2013 г.
    
Задачи:
    
    1. Знакомство с данными
    2. Проверка ошибок в данных и их предобработка
    3. Фильтрация данных
    4. Категория данных
    

### Описание данных

Данные `/datasets/new_games.csv` содержат информацию о продажах игр разных жанров и платформ, а также пользовательские и экспертные оценки игр:
- `Name` — название игры
- `Platform` — название платформы
- `Year of Release` — год выпуска игры
- `Genre` — жанр игры
- `NA sales` — продажи в Северной Америке (в миллионах проданных копий)
- `EU sales` — продажи в Европе (в миллионах проданных копий)
- `JP sales` — продажи в Японии (в миллионах проданных копий)
- `Other sales` — продажи в других странах (в миллионах проданных копий)
- `Critic Score` — оценка критиков (от 0 до 100)
- `User Score` — оценка пользователей (от 0 до 10)
- `Rating` — рейтинг организации ESRB (англ. Entertainment Software Rating Board). Эта ассоциация определяет рейтинг компьютерных игр и присваивает им подходящую возрастную категорию

### Содержимое проекта

 1. Загрузка данных и знакомство с ними
 2. Проверка ошибок в данных и их предобработка
 3. Фильтрация данных
 4. Категоризация данных
 5. Выводы
 

## Загрузка данных и знакомство с ними

Загрузим необходимые библиотеки Python и данные датасета `/datasets/new_games.csv`.


In [1]:
# Импортируем библиотеку pandas
import pandas as pd

In [2]:
# Выгружаем new_games.csv в датафрейм
df = pd.read_csv('https://code.s3.yandex.net/datasets/new_games.csv')

Познакомимся с данными: выведем первые строки и результат метода `info()`.


In [3]:
# Выгружаем первые 5 строк датафрейма
df.head(25)

,Name,Platform,Year of Release,Genre,NA sales,EU sales,JP sales,Other sales,Critic Score,User Score,Rating
0,Wii Sports,Wii,2006.0,Sports,41.36,28.96,3.77,8.45,76.0,8,E
1,Super Mario Bros.,NES,1985.0,Platform,29.08,3.58,6.81,0.77,NaN,NaN,NaN
2,Mario Kart Wii,Wii,2008.0,Racing,15.68,12.76,3.79,3.29,82.0,8.3,E
3,Wii Sports Resort,Wii,2009.0,Sports,15.61,10.93,3.28,2.95,80.0,8,E
4,Pokemon Red/Pokemon Blue,GB,1996.0,Role-Playing,11.27,8.89,10.22,1.00,NaN,NaN,NaN
5,Tetris,GB,1989.0,Puzzle,23.20,2.26,4.22,0.58,NaN,NaN,NaN
6,New Super Mario Bros.,DS,2006.0,Platform,11.28,9.14,6.5,2.88,89.0,8.5,E
7,Wii Play,Wii,2006.0,Misc,13.96,9.18,2.93,2.84,58.0,6.6,E
8,New Super Mario Bros. Wii,Wii,2009.0,Platform,14.44,6.94,4.7,2.24,87.0,8.4,E
9,Duck Hunt,NES,1984.0,Shooter,26.93,0.63,0.28,0.47,NaN,NaN,NaN


In [4]:
# Выгружаем информацию о дата фрейме
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16954 non-null  object 
 1   Platform         16956 non-null  object 
 2   Year of Release  16681 non-null  float64
 3   Genre            16954 non-null  object 
 4   NA sales         16956 non-null  float64
 5   EU sales         16956 non-null  object 
 6   JP sales         16956 non-null  object 
 7   Other sales      16956 non-null  float64
 8   Critic Score     8242 non-null   float64
 9   User Score       10152 non-null  object 
 10  Rating           10085 non-null  object 
dtypes: float64(4), object(7)
memory usage: 1.4+ MB


Датасет new_games.csv содержит 11 столбцов и 16956 строк, в которых представлена информация о продажах играх разных жанров, платформа, пользовательстких и экспертных оценках игр.

Изучим типы данных и их корректность:
- **Числовые значения с плавающей запятой (float64).** Четыре столбца имеют тип `float`:
   - `NA sales`, `Other sales` содержат значение продаж в Северной Америке и др. странах в млн. и представлены типом `float`. Это верное решение, так как цена в денежном выражении может включать дробные значения.
   - `Critic Score` сдержит оценки критиков от 0 до 100, поэтому нужно привести столбец к типу `float64`. 
   - `Year of Release` хранит информацию о годе выпуска игры. Для таких данных рекомендуется использовать тип `datetime64`, который облегчит выполнение временных вычислений и упростит анализ временных периодов.
- **Строковые данные (object).** Семь столбцов имеют тип данных `object`:
    - `Name`, `Platform` и `Genre` содержат строковую информацию (название игры, название платформы, жанр игры), что логично для текстовых данных. Здесь тип данных `object` подходит.
    - `EU sales` и `JP sales` содержат значение продаж в Европе и Японии в млн. Для таких данных рекомендуется использовать тип `float`.
    - `User Score`сдержит оценки пользователей от 0 до 10. При выводе значений заметно, что значения этого столбца — целые и дробные числа числа, поэтому тут подходит тип `float`. 
    - `Rating` хранит текстовые данные, но их можно рассматривать как категориальные признаки рейтинга игр. В этом случае можно использовать тип `category`, чтобы улучшить производительность и оптимизировать память.

После анализа типов данных видно, что часть столбцов корректно представлены. Для остальных стобцов требуется изменить тип данных в зависимости от хранимой в них информации.

Изучим наличие пропусков:

По RangeIndex видно, что всего в датафрейме 16956 строк, значит, пропуском будут считаться те столбцы, где количество ячеек меньше этого значения.

Пропуски есть в:
   - `Name`
   - `Year of Release`
   - `Genre`
   - `Critic Score`
   - `User Score`
   - `Rating`

Названия стобцов для удобства обращения к ним можно привести к snake case. Стоблцы названы корректно.

---

##  Проверка ошибок в данных и их предобработка


### Названия, или метки, столбцов датафрейма

Выведем на экран названия всех столбцов датафрейма и проверим их стиль написания.
Приведем все столбцы к стилю snake case. Названия должны быть в нижнем регистре, а вместо пробелов — подчёркивания.

In [5]:
# Выводим названия столбцо датафрейма
df.columns

Index(['Name', 'Platform', 'Year of Release', 'Genre', 'NA sales', 'EU sales',
       'JP sales', 'Other sales', 'Critic Score', 'User Score', 'Rating'],
      dtype='object')

In [6]:
# Приводим названия столбцов к стилю snake case и к нижнему регистру
df.columns = df.columns.str.replace(' ', '_').str.lower()
print(df.columns)

Index(['name', 'platform', 'year_of_release', 'genre', 'na_sales', 'eu_sales',
       'jp_sales', 'other_sales', 'critic_score', 'user_score', 'rating'],
      dtype='object')


### Типы данных

Если встречаются некорректные типы данных, предположим их причины.
При необходимости проведем преобразование типов данных.

Некорректные типы данных в столбцах возникают из-за особенностей CSV-файлов. Pandas по умолчанию помечает столбцы как object, если встречает нечисловые значения вроде пустых строк, "N/A", буквенных оценок или пропусков.

В числовых столбцах могут встретиться строковые значения, например `unknown` или другие. Приводемтакие столбцы к числовому типу данных, заменив строковые значения на пропуски.

In [7]:
# Изменяем тип данных
df['critic_score'] = pd.to_numeric(df['critic_score'], errors='coerce', downcast='float')
df['eu_sales'] = pd.to_numeric(df['eu_sales'], errors='coerce', downcast='float')
df['jp_sales'] = pd.to_numeric(df['jp_sales'], errors='coerce', downcast='float')
df['user_score'] = pd.to_numeric(df['user_score'], errors='coerce', downcast='float')
df['year_of_release'] = pd.to_datetime(df['year_of_release'], format='%Y', errors='coerce')
df['rating'] = df['rating'].astype('category')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   name             16954 non-null  object        
 1   platform         16956 non-null  object        
 2   year_of_release  16681 non-null  datetime64[ns]
 3   genre            16954 non-null  object        
 4   na_sales         16956 non-null  float64       
 5   eu_sales         16950 non-null  float32       
 6   jp_sales         16952 non-null  float32       
 7   other_sales      16956 non-null  float64       
 8   critic_score     8242 non-null   float32       
 9   user_score       7688 non-null   float32       
 10  rating           10085 non-null  category      
dtypes: category(1), datetime64[ns](1), float32(4), float64(2), object(3)
memory usage: 1.1+ MB


### Наличие пропусков в данных

- Посчитайте количество пропусков в каждом столбце в абсолютных и относительных значениях.


In [8]:
# Выводим количество пропущенных строк в датафрейме
df.isna().sum()

name                  2
platform              0
year_of_release     275
genre                 2
na_sales              0
eu_sales              6
jp_sales              4
other_sales           0
critic_score       8714
user_score         9268
rating             6871
dtype: int64

In [9]:
# Подсчитываем процент строк с пропусками
df.isna().sum() / len(df) * 100

name                0.011795
platform            0.000000
year_of_release     1.621845
genre               0.011795
na_sales            0.000000
eu_sales            0.035386
jp_sales            0.023590
other_sales         0.000000
critic_score       51.391838
user_score         54.659118
rating             40.522529
dtype: float64

Изучим данные с пропущенными значениями. Сделаем промежуточный вывод: для каких столбцов характерны пропуски и сколько их. Предположим, почему пропуски могли возникнуть.


Работа с пропусками:
   - `name` и `genre`имеют меньше 5% пропусков, поэтому строки можно удалить без потери данных
   - `year_of_release`имеет 1,6% пропусков, заменяем пропуски в столбце на медиану для каждой платформу и жанра
   - `critic_score`, `user_score`, `rating`игнорируем пропуски, так как их слишком много (от 40 до 50%) — удаление значительно повлияет на данные, лучше провести анализ с индикаторами NaN

Предположим возможные причины пропусков:
   - `name`, вероятно, случайные ошибки или ошибка загрузки данных в датафрейм, так как пропусков всего 2
   - `year_of_release`возможно, в первоисточнике информации из интернета не был указан год релиза
   - `critic_score`, `user_score`, `rating` это критерии являются необязательными, поэтому эта информация может не добавляться (например, игра непопулярная, не набрала достаточно оценок и т.д.).

Обработаем пропущенные значения. Для каждого случая выберем оптимальный вариант: заменить на определённое значение, оставить как есть или удалить.

In [10]:
# Удаляем пропуски в столбцах name и genre
df = df.dropna(subset=['name', 'genre'])
df.shape[0]

16954

In [11]:
# Заменяем пропуски в year_of_release на медиану для каждой платформу и жанра
def fill_year_median(row):
    if pd.isna(row['year_of_release']):
        # Находим строки с такой же комбинацией
        group = df[(df['platform'] == row['platform']) & 
                   (df['genre'] == row['genre'])]
        # Возвращаем медиану
        return group['year_of_release'].median()
    else:
        return row['year_of_release']

# Заполняем пропуски
df['year_of_release'] = df.apply(fill_year_median, axis=1)

# Удаляем строки, где year_of_release все еще NaN
df = df.dropna(subset=['year_of_release'])

print(f"Осталось строк после удаления: {df.shape[0]}")

/opt/conda/lib/python3.9/site-packages/numpy/lib/nanfunctions.py:1117: RuntimeWarning: Mean of empty slice
  return np.nanmean(a, axis, out=out, keepdims=keepdims)
/opt/conda/lib/python3.9/site-packages/numpy/lib/nanfunctions.py:1117: RuntimeWarning: Mean of empty slice
  return np.nanmean(a, axis, out=out, keepdims=keepdims)


Осталось строк после удаления: 16952


### Явные и неявные дубликаты в данных

Изучим уникальные значения в категориальных данных, например с названиями жанра игры, платформы, рейтинга и года выпуска. Проверьте, встречаются ли среди данных неявные дубликаты, связанные с опечатками или разным способом написания.
При необходимости проведем нормализацию данных с текстовыми значениями. Названия или жанры игр можно привести к нижнему регистру, а названия рейтинга — к верхнему.

In [12]:
# Выведем уникальные значения для столбцов с сортировкой, без пробелов и в нижнем регистре
df['name'] = df['name'].str.lower().str.strip()
sorted(df['name'].unique())

["'98 koshien",
 '.hack//g.u. vol.1//rebirth',
 '.hack//g.u. vol.2//reminisce',
 '.hack//g.u. vol.2//reminisce (jp sales)',
 '.hack//g.u. vol.3//redemption',
 '.hack//infection part 1',
 '.hack//link',
 '.hack//mutation part 2',
 '.hack//outbreak part 3',
 '.hack//quarantine part 4: the final chapter',
 '.hack: sekai no mukou ni + versus',
 '007 racing',
 '007: quantum of solace',
 '007: the world is not enough',
 '007: tomorrow never dies',
 '1 vs. 100',
 '1/2 summer +',
 '10 minute solution',
 '100 all-time favorites',
 '100 classic books',
 '100 classic games',
 '1000 cooking recipes from elle à table',
 '1001 touch games',
 '101-in-1 explosive megamix',
 '101-in-1 party megamix wii',
 '101-in-1 sports megamix',
 '101-in-1 sports party megamix',
 '1080°: teneighty snowboarding',
 '11eyes: crossover',
 '12-sai. honto no kimochi',
 '12-sai. koisuru diary',
 '12riven: the psi-climinal of integral',
 '15 days',
 '1701 a.d.',
 '18 wheeler: american pro trucker',
 '18 wheels of steel: ext

In [13]:
df['platform'] = df['platform'].str.upper().str.strip()
sorted(df['platform'].unique())

['2600',
 '3DO',
 '3DS',
 'DC',
 'DS',
 'GB',
 'GBA',
 'GC',
 'GEN',
 'GG',
 'N64',
 'NES',
 'NG',
 'PC',
 'PCFX',
 'PS',
 'PS2',
 'PS3',
 'PS4',
 'PSP',
 'PSV',
 'SAT',
 'SCD',
 'SNES',
 'TG16',
 'WII',
 'WIIU',
 'WS',
 'X360',
 'XB',
 'XONE']

In [14]:
df['genre'] = df['genre'].str.lower().str.strip()
sorted(df['genre'].unique())

['action',
 'adventure',
 'fighting',
 'misc',
 'platform',
 'puzzle',
 'racing',
 'role-playing',
 'shooter',
 'simulation',
 'sports',
 'strategy']

In [15]:
df['rating'].str.strip().unique()

array(['E', nan, 'M', 'T', 'E10+', 'K-A', 'AO', 'EC', 'RP'], dtype=object)

In [16]:
sorted(df['year_of_release'].unique())

[numpy.datetime64('1980-01-01T00:00:00.000000000'),
 numpy.datetime64('1981-01-01T00:00:00.000000000'),
 numpy.datetime64('1981-07-02T12:00:00.000000000'),
 numpy.datetime64('1982-01-01T00:00:00.000000000'),
 numpy.datetime64('1983-01-01T00:00:00.000000000'),
 numpy.datetime64('1984-01-01T00:00:00.000000000'),
 numpy.datetime64('1985-01-01T00:00:00.000000000'),
 numpy.datetime64('1986-01-01T00:00:00.000000000'),
 numpy.datetime64('1987-01-01T00:00:00.000000000'),
 numpy.datetime64('1988-01-01T00:00:00.000000000'),
 numpy.datetime64('1989-01-01T00:00:00.000000000'),
 numpy.datetime64('1990-01-01T00:00:00.000000000'),
 numpy.datetime64('1991-01-01T00:00:00.000000000'),
 numpy.datetime64('1992-01-01T00:00:00.000000000'),
 numpy.datetime64('1993-01-01T00:00:00.000000000'),
 numpy.datetime64('1994-01-01T00:00:00.000000000'),
 numpy.datetime64('1995-01-01T00:00:00.000000000'),
 numpy.datetime64('1995-07-02T12:00:00.000000000'),
 numpy.datetime64('1996-01-01T00:00:00.000000000'),
 numpy.datet

In [17]:
# Удаляем региональные пометки
df['name'] = df['name'].str.replace(r'\s*\([^)]*\)', '', regex=True)

# Удаляем указания на платформы в конце названий
platforms = r'(ds|wii|portable|3d|gc|psp|ps[0-9]?|xbox|gba|gb|nes|snes|n64|3ds|psvita|switch)$'
df['name'] = df['name'].str.replace(platforms, '', regex=True)

# Удаляем знаки пунктуации и специальные символы
df['name'] = df['name'].str.replace(r'[\[\]:;+&!]', '', regex=True)

# Удаляем двойные/множественные пробелы
df['name'] = df['name'].str.replace(r'\s+', ' ', regex=True)

print("Названия игр очищены от региональных пометок, платформ и лишних символов")

Названия игр очищены от региональных пометок, платформ и лишних символов


Дубликаты есть в столбце с названием игры `name`. Для него постарались выявить обозначения, создающие неявные дубликаты: сокращения названия страны, платформы, пунктуационные знаки. Для этой задачи использовала атрибут `regex`, чтобы не расписывать функцию с `if`? нашла этот метод в справке.
В других столбцах неявных дубликатов не обнаружено.

Проверим наличие явных дубликатов в данных.

In [18]:
# Считаем количество строк до удаления явных дубликатов
row_count = len(df)
print(f'Количество строчек до удаления явных дубликатов: {row_count}')

Количество строчек до удаления явных дубликатов: 16952


In [19]:
# Очищаем от явных дубликатов
df.drop_duplicates(keep='first', inplace=True)
row_count_without_duplicates = len(df)
print(f'Количество строчек после удаления явных дубликатов: {row_count_without_duplicates}')
print(df.head(50))

Количество строчек после удаления явных дубликатов: 16714
                                           name platform year_of_release  \
0                                    wii sports      WII      2006-01-01   
1                             super mario bros.      NES      1985-01-01   
2                                   mario kart       WII      2008-01-01   
3                             wii sports resort      WII      2009-01-01   
4                      pokemon red/pokemon blue       GB      1996-01-01   
5                                        tetris       GB      1989-01-01   
6                         new super mario bros.       DS      2006-01-01   
7                                      wii play      WII      2006-01-01   
8                        new super mario bros.       WII      2009-01-01   
9                                     duck hunt      NES      1984-01-01   
10                                   nintendogs       DS      2005-01-01   
11                            

Напишем промежуточный вывод: укажем количество найденных дубликатов и действия по их обработке.

In [20]:
# Количество удалённых дубликатов
print(f'Количество удалённых дубликатов: {row_count-row_count_without_duplicates}')

Количество удалённых дубликатов: 238


Явные дубликаты удалены по принципу: оставляем первую строчку с дублирующейся информацией.

In [21]:
# Количество удалённых строк, включая дубликаты, пропуски и ошибки
duplicates = 16956-row_count_without_duplicates
print(f'Количество удалённых строк: {duplicates}')

Количество удалённых строк: 242


In [22]:
# Процент удалённых строк, включая дубликаты, пропуски и ошибки, от общего числа
print(f'Процент удалённых строк: {round(duplicates / 16956 * 100, 2)}%')

Процент удалённых строк: 1.43%


В ходе обработки данных названия столбцов приведены к единому синтаксису, содержание столбцов написано в едином стиле в рамках каждого столбца, выбран оптимальный тип хранения данных и присвоен. Были выявлено закономерности, которые формировали неявные дубликаты и удалены. После обработанный датафрейм был очищен от явных дубликатов. Удалено 1,43% строк

---

## Фильтрация данных

Коллеги хотят изучить историю продаж игр в начале XXI века, и их интересует период с 2000 по 2013 год включительно. Отберите данные по этому показателю. Сохраните новый срез данных в отдельном датафрейме, например `df_actual`.

In [23]:
# Отфильтруем данные по году выпуска игры с 2000 по 2013 год
df_actual = df[(df['year_of_release'] >= '2000-01-01') & (df['year_of_release'] < '2014-01-01')]
print(df_actual)

                                                  name platform  \
0                                           wii sports      WII   
2                                          mario kart       WII   
3                                    wii sports resort      WII   
6                                new super mario bros.       DS   
7                                             wii play      WII   
...                                                ...      ...   
16947                     men in black ii alien escape       GC   
16949               woody woodpecker in crazy castle 5      GBA   
16950  score international baja 1000 the official game      PS2   
16952                                 lma manager 2007     X360   
16954                                   spirits spells      GBA   

      year_of_release     genre  na_sales   eu_sales  jp_sales  other_sales  \
0          2006-01-01    sports     41.36  28.959999      3.77         8.45   
2          2008-01-01    racing     1

---

## Категоризация данных
    
Проведем категоризацию данных.
Разделим все игры по оценкам пользователей и выделим такие категории: высокая оценка (от 8 до 10 включительно), средняя оценка (от 3 до 8, не включая правую границу интервала) и низкая оценка (от 0 до 3, не включая правую границу интервала).

In [24]:
# Категоризируем данные по пользовательской оценке
def categorize_user_score(score):
    if pd.isna(score):
        return 'Нет оценки'
    elif score >= 8:
        return 'Высокая оценка'
    elif score >= 3:
        return 'Средняя оценка'
    else:
        return 'Низкая оценка'

# Применяем категоризацию
df_actual['user_score_category'] = df_actual['user_score'].apply(categorize_user_score)

print(df_actual)

                                                  name platform  \
0                                           wii sports      WII   
2                                          mario kart       WII   
3                                    wii sports resort      WII   
6                                new super mario bros.       DS   
7                                             wii play      WII   
...                                                ...      ...   
16947                     men in black ii alien escape       GC   
16949               woody woodpecker in crazy castle 5      GBA   
16950  score international baja 1000 the official game      PS2   
16952                                 lma manager 2007     X360   
16954                                   spirits spells      GBA   

      year_of_release     genre  na_sales   eu_sales  jp_sales  other_sales  \
0          2006-01-01    sports     41.36  28.959999      3.77         8.45   
2          2008-01-01    racing     1

/tmp/ipykernel_92/1665730383.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_actual['user_score_category'] = df_actual['user_score'].apply(categorize_user_score)


Разделим все игры по оценкам критиков и выделим такие категории: высокая оценка (от 80 до 100 включительно), средняя оценка (от 30 до 80, не включая правую границу интервала) и низкая оценка (от 0 до 30, не включая правую границу интервала).

In [25]:
# Категоризируем данные по оценкам критиков
def categorize_critic_score(score):
    if pd.isna(score):
        return 'Нет оценки'
    elif score >= 80:
        return 'Высокая оценка'
    elif score >= 30:
        return 'Средняя оценка'
    else:
        return 'Низкая оценка'

# Применяем категоризацию
df_actual['critic_score_category'] = df_actual['critic_score'].apply(categorize_critic_score)

print(df_actual)

                                                  name platform  \
0                                           wii sports      WII   
2                                          mario kart       WII   
3                                    wii sports resort      WII   
6                                new super mario bros.       DS   
7                                             wii play      WII   
...                                                ...      ...   
16947                     men in black ii alien escape       GC   
16949               woody woodpecker in crazy castle 5      GBA   
16950  score international baja 1000 the official game      PS2   
16952                                 lma manager 2007     X360   
16954                                   spirits spells      GBA   

      year_of_release     genre  na_sales   eu_sales  jp_sales  other_sales  \
0          2006-01-01    sports     41.36  28.959999      3.77         8.45   
2          2008-01-01    racing     1

/tmp/ipykernel_92/3183801276.py:13: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_actual['critic_score_category'] = df_actual['critic_score'].apply(categorize_critic_score)


После категоризации данных проверим результат: сгруппируем данные по выделенным категориям и посчитаем количество игр в каждой категории.

In [26]:
# Посчитаем количество строк для каждой категории `user_score_category`
count_user_score_category = df_actual['user_score_category'].value_counts()
print(count_user_score_category)

Нет оценки        6416
Средняя оценка    4159
Высокая оценка    2328
Низкая оценка      119
Name: user_score_category, dtype: int64


In [27]:
# Посчитаем количество строк для каждой категории `critic_score_category`
count_critic_score_category = df_actual['critic_score_category'].value_counts()
print(count_critic_score_category)

Нет оценки        5703
Средняя оценка    5537
Высокая оценка    1724
Низкая оценка       58
Name: critic_score_category, dtype: int64


Выделим топ-7 платформ по количеству игр, выпущенных за весь актуальный период.

In [28]:
#Gjlcxbnftv количество игр для каждой платформы и отфильруем от болшего к меньшему
df_actual.groupby('platform')['platform'].count().sort_values(ascending = False)

platform
PS2     2161
DS      2150
WII     1309
PSP     1196
X360    1152
PS3     1112
XB       824
GBA      822
PC       783
GC       556
3DS      307
PS       274
PSV      134
WIIU      74
N64       71
DC        31
GB        27
XONE      19
PS4       16
WS         4
Name: platform, dtype: int64

Топ-7 популярных платформ для игр за период с 2000 по 2013 года включительно:
   1. `PS2`
   2. `DS`
   3. `WII`   
   4. `PSP`
   5. `X360` 
   6. `PS3`
   7. `XB`

---

## Итоговый вывод

По итогам работы:
   - названия столбцов приведены к нижнему регистру и shake_case
   - для каждого столбца в датафрейме подобран оптимальный и подходящий тип хранения данных
   - выполнена предобработка данных
   - удалены пропуски в данных, которые были критичны для корректного анализа
   - заполнены пропуски в стобце `year_of_release` медианным значением
   - оставлены пропуски в столбцах с рейтинговой оценкой, так как они допустимы
   - изучена логика возникновения неявных дубликатов и на основании этого данные очищены от симфолов и слов, создающих неявные дубликаты
   - датафрейм очищен от явных дубликатов
   - количество удалённых строк: 242 или 1,43%
   - данные отфильтрованы по году выпуска игры с 2000 по 2013 год включительно и помещены в новый датафрейм `df_actual`
   - пользовательские и экспертные оценки категоризированы по категориям `Высокая оценка`, `Средняя оценка`, `Низкая оценка` и нет `Нет оценки`и созданы новые столбцы `user_score rating` `user_score_category`
   - мнение пользователей и экспертов по оценки игр совпадает
   - самые популярные игровые платформы: `PS2`, `DS`, `WII`, `PSP`, `X360`, `PS3`, `XB`
   
   